public/app/features/transformers/calculateHeatmap/models.gen.ts

```diff
@@ -15,4 +15,8 @@ export interface HeatmapCalculationOptions {
   yAxis?: HeatmapCalculationAxisConfig;
   xAxisField?: string; // name of the x field
   encoding?: DataFrameType.HeatmapBuckets | DataFrameType.HeatmapScanlines;
+
+  // these typically come from dashboard TimeRange timestamps
+  xFrom: number;
+  xTo: number;
```

Comment:
Maybe these should be an additional argument to the function -- the structure defined here will end up as an option in the saved JSON schema, so better to keep it seperate